src/types/SubChunkPacketEntryCommon.php

```diff
@@ -78,5 +92,17 @@ public function write(PacketSerializer $out, bool $cacheEnabled) : void{
 			$out->putByte(SubChunkPacketHeightMapType::DATA);
 			$heightMapData->write($out);
 		}
+
+		if($this->renderHeightMap === null){
+			$out->putByte(SubChunkPacketHeightMapType::ALL_COPIED);
```

Comment:
Are you sure this is correct? What happens if NO_DATA is used?

Reply:
Absolutely no clue, it's very hard for me to test.